configure.ac

```diff
@@ -872,6 +872,96 @@ AM_CONDITIONAL([OPENTHREAD_ENABLE_RAW_LINK_API], [test "${enable_raw_link_api}"
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_RAW_LINK_API],[${OPENTHREAD_ENABLE_RAW_LINK_API}],[Define to 1 if you want to enable raw link-layer API])
 
 #
+# Software ACK Timeout Logic
+#
+
+AC_ARG_ENABLE(sw_ack_timeout,
```

Comment:
I agree the makefiles are neat and clean with this approach.  But I am afraid we may need to add many new config features in `configure.ac`...
 
I think in long-term we will probably need a separate config-file for each platform (we will need it if we want to change the number/size of different buffers, etc... which are platform dependant )... And we already have it for nrf52840.  We can start adding it now for each platform. 

Thoughts?